src/Compilers/Core/CodeAnalysisTest/PEWriter/BlobTests.cs

```diff
@@ -856,5 +856,91 @@ public void Pooled()
             builder1.Free();
             builder2.Free();
         }
+
+        /// <summary>
+        /// This stream requires at least two calls to Read() to read all elements.
+        /// </summary>
+        private class TwoReadStream : Stream
+        {
+            public static byte[] BackingData { get; } = { 1, 2, 3, 4, 5 };
+
+            public override bool CanRead
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override bool CanSeek
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override bool CanWrite
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override long Length => BackingData.Length;
+
+            public override long Position { get; set; } = 0;
+
+            public override void Flush()
+            {
+                throw new NotImplementedException();
+            }
+
+            private bool _firstRead = true;
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+
+                // No matter what the client asks for, only give them half the first time
+                // Unless they only ask for 1 byte
+                if (_firstRead)
+                {
+                    count = (int)Math.Ceiling(count / 2.0);
```

Comment:
I think the behavior of the class should make sense on its own. Either the client should drive it, for example, by calling a special method prescribing what the sequence of Read calls should return, or behavior should be hard-coded (first, I give 2 bytes, then 3, then 0, etc.) The fact that the method never returns 0 clearly indicates a test hole.